Scripts/GetFieldsByIncidentType/GetFieldsByIncidentType.py

```diff
@@ -0,0 +1,79 @@
+import demistomock as demisto
+from CommonServerPython import *
+from pprint import pformat
+
+
+def get_field_by_long_name(fields, name):
+    for field in fields:
+        if field['name'] == name:
+            return field
+    return None
+
+
+def get_field_by_short_name(fields, name):
+    for field in fields:
+        if field['cliName'] == name:
+            return field
+    return None
+
+
+def main():
+    # get incident fields
+    res = demisto.executeCommand('demisto-api-get', {'uri': '/incidentfields'})
+    if is_error(res):
+        return_error(res[0]['Contents'])
+
+    fields = res[0]['Contents']['response']
+
+    # 'fields' contains non-incident fields, as well, so let's make a version containing only incident fields
+    incident_fields = [field for field in fields if field['id'].startswith('incident_')]
+
+    # get arguments
+    args = demisto.args()
+
+    incident_type = args['incident_type']
+
+    exclude_system = False
+    if 'custom' in args and argToBoolean(args['custom']) is True:
+        exclude_system = True
+
+    name_key = 'name'
+    if 'short_names' in args and argToBoolean(args['short_names']) is True:
+        name_key = 'cliName'
+
+    explicit_only = False
+    if 'explicit_only' in args and argToBoolean(args['explicit_only']) is True:
+        explicit_only = True
```

Comment:
It means that if the field has 'associate to all' checked, it will be excluded from the results.  I've updated the description to read "If  true, only fields which are explicitly associated to the incident type will be included (i.e. excludes fields with the "associate to any" box ticked)"

Really?  I think the args.get() convention is less readable than the way I wrote it.

Also, doesn't args.get() fail to normalise strings into booleans?  That's why I pass it through argToBoolean().